code/game/machinery/hologram.dm

```diff
@@ -68,199 +95,324 @@ var/list/holopads = list()
 	default_deconstruction_crowbar(P)
 
 
-/obj/machinery/hologram/holopad/attack_hand(var/mob/living/carbon/human/user) //Carn: Hologram requests.
+/obj/machinery/hologram/holopad/proc/CheckCallClose()
+	for(var/I in holo_calls)
+		var/datum/holocall/HC = I
+		if(usr == HC.eye)
+			HC.Disconnect(HC.calling_holopad)	//disconnect via clicking the called holopad
+			return TRUE
+	return FALSE
+
+/obj/machinery/hologram/holopad/attack_hand(mob/living/carbon/human/user) //Carn: hologram requests.
 	if(!istype(user))
 		return
-	if(alert(user,"Would you like to request an AI's presence?",,"Yes","No") == "Yes")
-		if(last_request + 200 < world.time) //don't spam the AI with requests you jerk!
+
+	if(!CheckCallClose())
+		return
+
+	if(outgoing_call || user.incapacitated() || !(stat & NOPOWER))

```

Comment:
`|| (stat & NOPOWER)` is what you probably want here, since otherwise it will meet this condition while it DOES have power, thus returning prematurely and not working beyond this condition.